cfme/tests/v2v/test_v2v_cancel_migrations.py

```diff
@@ -4,97 +4,92 @@
+                                            mapping_data_multiple_vm_obj_single_datastore):
     """
     Polarion:
-        assignee: ytale
-        initialEstimate: 1/4h
+        assignee: sshveta
+        initialEstimate: 1/2h
+        caseimportance: medium
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.10
         casecomponent: V2V
+        testSteps:
+            1. Add source and target provider
+            2. Create infra map and migration plan
+            3. Start migration plan and cancel.
     """
-    # TODO: Improve this test to cover cancel operation at various stages in migration.
     # This test will make use of migration request details page to track status of migration
-    infrastructure_mapping_collection = appliance.collections.v2v_mappings
-    mapping = infrastructure_mapping_collection.create(
-        form_data_multiple_vm_obj_single_datastore.form_data)
+    cancel_migration_after_percent = 50
+    infrastructure_mapping_collection = appliance.collections.v2v_infra_mappings
+    mapping_data = mapping_data_multiple_vm_obj_single_datastore.infra_mapping_data
+    mapping = infrastructure_mapping_collection.create(**mapping_data)
 
     @request.addfinalizer
     def _cleanup():
         infrastructure_mapping_collection.delete(mapping)
 
-    migration_plan_collection = appliance.collections.v2v_plans
+    migration_plan_collection = appliance.collections.v2v_migration_plans
     migration_plan = migration_plan_collection.create(
-        name="plan_{}".format(fauxfactory.gen_alphanumeric()), description="desc_{}"
-        .format(fauxfactory.gen_alphanumeric()), infra_map=mapping.name,
-        vm_list=form_data_multiple_vm_obj_single_datastore.vm_list, start_migration=True)
-    # as migration is started, try to track progress using migration plan request details page
-    view = appliance.browser.create_view(navigator.get_class(migration_plan_collection,
-                                                            'All').VIEW.pick())
-    wait_for(func=view.progress_card.is_plan_started, func_args=[migration_plan.name],
-        message="migration plan is starting, be patient please", delay=5, num_sec=150,
-        handle_exception=True, fail_cond=False)
-    view.progress_card.select_plan(migration_plan.name)
-    view = appliance.browser.create_view(navigator.get_class(migration_plan_collection,
-                                                             'Details').VIEW, wait='10s')
-    request_details_list = view.migration_request_details_list
+        name="plan_{}".format(fauxfactory.gen_alphanumeric()),
+        description="desc_{}".format(fauxfactory.gen_alphanumeric()),
+        infra_map=mapping.name,
+        target_provider=provider,
+        vm_list=mapping_data_multiple_vm_obj_single_datastore.vm_list,
+    )
+
+    assert migration_plan.wait_for_state("Started")
+    request_details_list = migration_plan.get_plan_vm_list(wait_for_migration=False)
```

Comment:
@sshveta  we don't have `wait_for_migration=False` args for `get_plan_vm_list` can you check again?

Reply:
It is there @Yadnyawalkya

Reply:
https://github.com/ManageIQ/integration_tests/blob/master/cfme/v2v/migration_plans.py#L443

Reply:
My bad! I think I am working with very old code. :D